test/src/micro/attributes_test.dart

```diff
@@ -72,3 +90,55 @@ class AttributesTest extends PolymerElement {
 
   AttributesTest.created() : super.created();
 }
+
+enum Foobar { foo, bar }
+
+@PolymerRegister(_serializeAttributesName, hostAttributes: const {
+  'host-string': 'string',
+  'host-num': 2,
+  'host-bool': true,
+  //'host-map': const {'hello': 'world'},
+  //'host-list': const ['hello', 'world'],
+  // TODO(jakemac): Do we need to support this?
+//  'host-date-time': new DateTime(1987, 07, 18),
+  'host-foobar': Foobar.bar
+})
+class SerializedAttributesTest extends PolymerElement with PolymerSerialize {
```

Comment:
I can give it a go. I stayed away from it because of the Polymer.js restriction. Wasn't sure if it would actually work.